ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts

```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
+                          '-01T00:00',
+                activeTo:
+                    activeToDateFormat.getMonth() === 0
+                        ? activeToDateFormat.getFullYear() - 1 + '-12-01T00:00'
+                        : activeToDateFormat.getFullYear() +
+                          '-' +
+                          String(activeToDateFormat.getMonth()).padStart(2, '0') +
+                          '-01T00:00'
+            };
+        }
+    }
+
+    public getDatesWhenMovingWithWeekPeriodClicked(
+        activeFrom: string,
+        activeTo: string,
+        isForward: boolean
+    ): {activeFrom: any; activeTo: any} {
+        if (isForward) {
+            return {
+                activeFrom: moment(activeFrom, 'YYYY-MM-DDTHH:mm').add(7, 'days').format('YYYY-MM-DDTHH:mm'),
+                activeTo: moment(activeTo, 'YYYY-MM-DDTHH:mm').add(7, 'days').format('YYYY-MM-DDTHH:mm')
+            };
+        } else {
+            return {
+                activeFrom: moment(activeFrom, 'YYYY-MM-DDTHH:mm').subtract(7, 'days').format('YYYY-MM-DDTHH:mm'),
+                activeTo: moment(activeTo, 'YYYY-MM-DDTHH:mm').subtract(7, 'days').format('YYYY-MM-DDTHH:mm')
+            };
+        }
+    }
+
+    public getDatesWhenMoving(
+        activeFrom: string,
+        activeTo: string,
+        isForward: boolean
+    ): {activeFrom: any; activeTo: any} {
+        const activeFromDateFormat = new Date(activeFrom);
+        const activeToDateFormat = new Date(activeTo);
+
+        let newActiveFromDate: string, newActiveToDate: string;
+        if (isForward) {
+            newActiveFromDate = new Date(
+                activeFromDateFormat.getTime() + (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+            newActiveToDate = new Date(
+                activeToDateFormat.getTime() + (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+        } else {
+            newActiveFromDate = new Date(
+                activeFromDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+            newActiveToDate = new Date(
+                activeToDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+        }
+        return {
+            activeFrom: this.formatDateLocalStringForInput(newActiveFromDate),
+            activeTo: this.formatDateLocalStringForInput(newActiveToDate)
+        };
+    }
+
+    public formatDateLocalStringForInput(dateLocalString: string) {
+        return (
+            dateLocalString.substring(6, 10) +
+            '-' +
+            dateLocalString.substring(3, 5) +
+            '-' +
+            dateLocalString.substring(0, 2) +
+            'T' +
+            dateLocalString.substring(11, 20)
```

Comment:
The format seems not consistent with other methods , the substring (11,20) leads to add seconds in the string where in other method you do not have  (YYYY-MM-DDTHH:MM)

Reply:
I guess you mean "you do not have (YYYY-MM-DDTHH:MM:ss)"

Reply:
Done.